assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -41,10 +47,57 @@ interface Store {
 	ref?: HTMLElement;
 }
 
+interface Event {
+	keyCode: number;
+}
+
 type SelectorsStore = Pick< Store, 'context' | 'selectors' | 'ref' >;
 
 interactivityApiStore( {
 	state: {},
+	effects: {
+		woocommerce: {
+			keyboardAccess: ( store: Store ) => {
+				const { context, actions } = store;
+				let allowNavigation = true;
+
+				const handleKeyEvents = ( event: Event ) => {
+					if (
+						! allowNavigation ||
+						! context.woocommerce.isDialogOpen
+					) {
+						return;
+					}
+
+					// Disable navigation for a brief period to prevent spamming
+					allowNavigation = false;
+
+					setTimeout( () => {
+						allowNavigation = true;
+					}, 300 );
+
+					// Check if the esc key is pressed.
+					if ( event.keyCode === 27 ) {
+						context.woocommerce.isDialogOpen = false;
+					}
+
+					// Check if left arrow key is pressed.
+					if ( event.keyCode === 37 ) {
+						actions.woocommerce.handlePreviousImageButtonClick(
+							store
+						);
+					}
+
+					// Check if right arrow key is pressed.
+					if ( event.keyCode === 39 ) {
+						actions.woocommerce.handleNextImageButtonClick( store );
+					}
+				};
+
+				document.addEventListener( 'keydown', handleKeyEvents );
```

Comment:
Should we remove this event listener when the gallery is not open?

Reply:
Yes we probably should but I haven't found an elegant way to do that in this context. Any suggestions?

Reply:
> For some reasons it is only showing 3 images in the gallery for me. Is this expected?
Yes that's expected and being worked on else where.